torch/utils/launch.py

```diff
@@ -0,0 +1,805 @@
+
+    parser.add_argument('--distributed', action='store_true', default=False,
+                        help='Enable distributed training.')
+    parser.add_argument("-m", "--module", default=False, action="store_true",
+                        help="Changes each process to interpret the launch script "
+                             "as a python module, executing with the same behavior as"
+                             "'python -m'.")
+
+    parser.add_argument("--no_python", default=False, action="store_true",
+                        help="Do not prepend the --program script with \"python\" - just exec "
+                             "it directly. Useful when the script is not a Python script.")
+
+    add_memory_allocator_params(parser)
+    add_kmp_iomp_params(parser)
+
+    add_distributed_training_params(parser)
+    add_multi_instance_params(parser)
+    # positional
+    parser.add_argument("program", type=str,
+                        help="The full path to the proram/script to be launched. "
+                             "followed by all the arguments for the script")
+
+    # rest from the training program
+    parser.add_argument('program_args', nargs=REMAINDER)
+    return parser.parse_args()
+
+def main():
+
+    env_before = set(os.environ.keys())
+    if platform.system() == "Windows":
+        raise RuntimeError("Windows platform is not supported!!!")
+
+    args = parse_args()
+    if args.log_path:
+        path = os.path.dirname(args.log_path if args.log_path.endswith('/') else args.log_path + '/')
+        if not os.path.exists(path):
+            os.makedirs(path)
+        args.log_path = path
+
+        args.log_file_prefix = '{}_{}'.format(args.log_file_prefix, datetime.now().strftime("%Y%m%d%H%M%S"))
+        fileHandler = logging.FileHandler("{0}/{1}_instances.log".format(args.log_path, args.log_file_prefix))
+        logFormatter = logging.Formatter(format_str)
+        fileHandler.setFormatter(logFormatter)
+        logger.addHandler(fileHandler)
+
+    if args.distributed and args.multi_instance:
+        raise RuntimeError("Either args.distributed or args.multi_instance should be set")
+
+    if args.latency_mode and args.throughput_mode:
+        raise RuntimeError("Either args.latency_mode or args.throughput_mode should be set")
+
+    if args.nnodes > 1:
+        args.distributed = True
+
+    if not args.no_python and not args.program.endswith(".py"):
+        logger.error("For non Python script, you should use '--no_python' parameter.")
+        exit()
+
+    # Verify LD_PRELOAD
+    if "LD_PRELOAD" in os.environ:
+        lst_valid = []
+        tmp_ldpreload = os.environ["LD_PRELOAD"]
+        for item in tmp_ldpreload.split(":"):
+            matches = glob.glob(item)
+            if len(matches) > 0:
+                lst_valid.append(item)
+            else:
+                logger.warning("{} doesn't exist. Removing it from LD_PRELOAD.".format(item))
```

Comment:
We are in Python-3 world, please use f-strings
```suggestion
                logger.warning(f"{item} doesn't exist. Removing it from LD_PRELOAD.")
```

Reply:
Sure.